### PR TITLE
Convert the C test suite to lang_tester.

### PR DESCRIPTION
**Pull request by vext01**

### CODE REVIEW

File: c_tests/tests/blockmap.c [hunk 1]
```diff
@@ -1,3 +1,6 @@
+// Compiler:
```
Comment (ltratt) [inline c_tests/tests/blockmap.c:1]: Shouldn't we at least check for `status: success`? At the moment these tests don't test anything at all.

Comment (vext01) [reply c_tests/tests/blockmap.c:1]: Oops. Yes, that was my intention!

Comment (vext01) [reply c_tests/tests/blockmap.c:1]: Docs say:

> If not specified, defaults to success.

Assuming that's correct?

Comment (vext01) [reply c_tests/tests/blockmap.c:1]: I put an `assert(0)` in the test and it flagged as a failure, so I think the docs are correct.

Comment (ltratt) [reply c_tests/tests/blockmap.c:1]: I stand corrected!

**Comment by vext01** [comment]
bors try

**Comment by vext01** [comment]
OK to squash?

**Comment by ltratt** [comment]
Please squash.

**Comment by vext01** [comment]
splat

**Pull request merged**